### DISCUSSION TITLE
有没有使用xray核心的交流下

### DISCUSSION THREAD

**Discussion by z1073** [v2rayA/v2rayA#1031] [Q&A]
我主要想使用新出的那个reality协议，使用docker镜像安装的v2rayA，用V2RAYA_V2RAY_BIN变量参数指定了xray核心的位置，同时也把xray目录挂载到了docker镜像里，但是进入v2rayA管理页面一直提示找不到xray核心，目录也给足了权限，不知道问题在哪，有没有使用xray核心的朋友讲讲你们怎么切换的核心

![image](https://github.com/v2rayA/v2rayA/assets/50789523/813faef6-b242-4edd-8819-6f36a9f4b7e7)

2023/08/13 13:21:07.341 [A] [main.go:26] V2Ray binary is /etc/v2raya/xray
2023/08/13 13:21:07.342 [A] [main.go:26] V2Ray asset directory is /run/user/0/v2raya
2023/08/13 13:21:07.342 [A] [main.go:26] v2rayA working directory is /root
2023/08/13 13:21:07.342 [A] [main.go:26] v2rayA configuration directory is /etc/v2raya
2023/08/13 13:21:07.343 [A] [main.go:26] Golang: go1.20.7
2023/08/13 13:21:07.343 [A] [main.go:26] OS: linux
2023/08/13 13:21:07.343 [A] [main.go:26] Arch: amd64
2023/08/13 13:21:07.343 [A] [main.go:26] Lite: false
2023/08/13 13:21:07.343 [A] [main.go:26] Version: 2.1.0
2023/08/13 13:21:07.343 [A] [main.go:26] Starting...
2023/08/13 13:21:07.449 [E] [main.go:27] failed to start v2ray-core: fork/exec /etc/v2raya/xray: permission denied
2023/08/13 13:21:07.469 [A] [index.go:116] v2rayA is listening at http://127.0.0.1:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://192.168.1.199:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://172.17.0.1:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://172.18.0.1:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://172.19.0.1:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://[::1]:2028
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://[fe80::7385:c2ff:fe82:1ffe]:2023
2023/08/13 13:21:07.470 [A] [index.go:116] v2rayA is listening at http://[fe80::42:afff:fefc:617]:202

**docker配置：**
docker run -d \
  --restart=always \
  --privileged \
  --network=host \
  --name v2raya \
  -e V2RAYA_LOG_FILE=/tmp/v2raya.log \
  -e V2RAYA_V2RAY_BIN=/etc/v2raya/xray \
  -e V2RAYA_ADDRESS=0.0.0.0:2023
  -v /etc/v2raya/xray:/etc/v2raya/xray \
  -v /lib/modules:/lib/modules:ro \
  -v /etc/resolv.conf:/etc/resolv.conf \
  -v /etc/v2raya:/etc/v2raya \
  mzz2017/v2raya

**Comment by RuiKylin** [comment]
docker使用的话，现在使用作者的镜像可以解决这个问题，但是gfwlist更新问题依旧。

使用dockerfile自构建的镜像就会出现你这样的问题。

**Comment by jackect** [comment]
你应该再配置一下XRAY_LOCATION_ASSET这个环境变量，指定到geoip.dat那个路径

**Comment by standin000** [comment]
xray内核不能使用负载均衡
